readme.md

```diff
@@ -188,6 +183,47 @@ mbo.finalize().then(function(mediaBlobs) {
 });
 ```
 
+### Error Handling in finalize
+When *[finalize()](#finalize-method)* is called, the User Agent will perform these basic checks for the operations that are batched. This error checking should be done before executing any of the operations.
+
+For *[trim()](#trim-method)*
+1. Let *O* represent the blob to be trimmed
+2. If *startTime* is less than 0 **OR** *endTime* is greater than the *O*.duration **OR** *startTime* is greater than the *endTime*:
+    * create a new [ErrorEvent](https://html.spec.whatwg.org/multipage/webappapis.html#errorevent)
+    * report the exception event 
+    * break
+
+For *[split()](#split-method)*
+1. Let *O* represent the blob to be split
+2. If *time* is less than 0 **OR** is greater than *O*.duration **OR** this is not the last operation before finalize() was called
+    * create a new [ErrorEvent](https://html.spec.whatwg.org/multipage/webappapis.html#errorevent)
+    * report the exception event 
+    * break
+
+For *[concat()](#concat-method)*
+1. Let *m1* represent the first *MediaBlob* which will be the *MediaBlob* from the *MediaBlobOperation* that has the *concat* method called upon
+2. Let *m2* represent the *MediaBlob* that is passed in to *concat* method to be concatenated with *m1*
+3. If the mimeType of m1 does not equal the mimeType of m2:
+    * create a new [ErrorEvent](https://html.spec.whatwg.org/multipage/webappapis.html#errorevent)
+    * report the exception event
+    * break
+
+The ErrorEvent.message will contain:
+1. Operation name
+2. The sequence number indicating the position of the operation
+3. Actual error message
+
+Example:
+```
+let mbo = new MediaBlobOperation(new MediaBlob(blob));
+mbo.trim(0,5000);  // Trim from 0 to 5 secs
+mbo.split(7000);  // Split the MediaBlob at 7 secs
+mbo.finalize().then(function(mediaBlobs) {
```

Comment:
For this example, I'd expect the promise to reject, which would invoke the second parameter in then(), which is the reject handler.  See:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

Alternatively, developers can add .catch() to the end of a promise change to capture errors.

mbo.finalize().then(function(mediaBlobs) { }).catch(function(error) {  });

Reply:
Yes, missed that. I will update this.

Reply:
Updated.